src/MultiSigWallet.sol

```diff
@@ -142,10 +182,17 @@ contract MultiSigWallet is Common{
 
     function publishTransaction(uint _txIndex) external {
         //TODO: check for sufficient balance.
-        require(transactions[_txIndex].confirmationsDone>= numOfConfirmationsRequired, "Need more approvals");
-        (bool sent, ) = transactions[_txIndex].to.call{value: transactions[_txIndex].amount}(transactions[_txIndex].data);
-        require(sent, "Failed to send Ether");
-        transactions[_txIndex].executed=true;
+        Transaction memory transaction = transactions[_txIndex];
+        require(transaction.confirmationsDone>= numOfConfirmationsRequired, "Need more approvals");
+        if(transaction.transactionType == 1){
+            (bool sent, ) = transaction.to.call{value: transactions[_txIndex].amount}(transactions[_txIndex].data);
+            require(sent, "Failed to send Ether");
+            transactions[_txIndex].executed=true;
+        }else{
+            bool sent = IERC20(transaction.token).safeTransferFrom(owner, transaction.to, transaction.amount);
```

Comment:
Why do we want to safeTransferfrom here? We do safe transferFrom when user wants to deposit money  in the smartWallet but while doing transaction user already has the assets in his smart wallet right. ```IERC20(transaction.token).safeTransfer( transaction.to, transaction.amount);``` some thing like this should be fine.